driver/razerchromacommon.c

```diff
@@ -137,7 +137,45 @@ struct razer_report razer_chroma_standard_get_led_rgb(unsigned char variable_sto
     return report;
 }
 
+/*
+ * sets fan speed
+ *
+ */
+struct razer_report razer_chroma_set_fan_speed(unsigned char fan_id, unsigned char fan_speed)
+{
+    struct razer_report report = get_razer_report(0x0d, 0x01, 0x03);
+    report.arguments[0] = 0x00;   //frame_id: fixed value
+    report.arguments[1] = fan_id; //maybe fans can be set separately by id?
+    report.arguments[2] = fan_speed;
+
+    return report;
+}
+
+
+struct razer_report razer_chroma_get_fan_speeed(unsigned char fan_id)
+{
+    struct razer_report report = get_razer_report(0x0d, 0x84, 0x03);
+    report.transaction_id.id = 0x3F;
 
+    report.arguments[0] = 0x00;
+    report.arguments[1] = fan_id;
+
+    return report;
+}
+
+/*
+ * sets fan mode - manual or automatic
+ */
+struct razer_report razer_chroma_set_fan_mode(unsigned char fan_id, unsigned char fan_mode)
+{
+    struct razer_report report = get_razer_report(0x0d, 0x02, 0x04);
+    report.arguments[0] = 0x00;   //frame_id: fixed value
+    report.arguments[1] = fan_id; //maybe fans can be set separately by id?
+    report.arguments[2] = 0x00;   // unknown observed: 0x00
```

Comment:
This argument is for power mode of the CPU,
0 - balanced
1 - gaming
2 - creator
4 - custom
with 4 goes CPU boost and GPU boost power

Reply:
thx! added now